lib/active_merchant/billing/credit_card_methods.rb

```diff
@@ -103,37 +128,18 @@ def valid_number?(number)
             valid_checksum?(number)
         end
 
-        # Regular expressions for the known card companies.
-        #
-        # References:
-        # - http://en.wikipedia.org/wiki/Credit_card_number
-        # - http://www.barclaycardbusiness.co.uk/information_zone/processing/bin_rules.html
         def card_companies
-          CARD_COMPANIES
+          CARD_COMPANIES.keys
         end
 
         # Returns a string containing the brand of card from the list of known information below.
-        # Need to check the cards in a particular order, as there is some overlap of the allowable ranges
-        #--
-        # TODO Refactor this method. We basically need to tighten up the Maestro Regexp.
-        #
-        # Right now the Maestro regexp overlaps with the MasterCard regexp (IIRC). If we can tighten
-        # things up, we can boil this whole thing down to something like...
-        #
-        #   def brand?(number)
-        #     return 'visa' if valid_test_mode_card_number?(number)
-        #     card_companies.find([nil]) { |brand, regexp| number =~ regexp }.first.dup
-        #   end
-        #
         def brand?(number)
           return 'bogus' if valid_test_mode_card_number?(number)
 
-          card_companies.reject { |c,p| c == 'maestro' }.each do |company, pattern|
-            return company.dup if number =~ pattern
+          CARD_COMPANIES.each do |company, func|
+            return company.dup if func[number]
```

Comment:
I'd normally expect to see a lambda invoked via `.call`, so `func.call(number)`. Is there a benefit to invoking the function as `func[number]`?

Reply:
Nope, just me not knowing what's Rubyish.

Reply:
Done.